wallet/src/storage.rs

```diff
@@ -0,0 +1,11 @@
+use serde::{Deserialize, Serialize};
+
+use penumbra_crypto::keys;
+
+/// The contents of the wallet file that share a spend authority.
+#[derive(Serialize, Deserialize)]
+pub struct Wallet {
+    pub spend_seed: keys::SpendSeed,
+    // Store the last diversifier index in use.
+    pub last_used_diversifier_index: keys::DiversifierIndex,
```

Comment:
I was on the fence about making this `Vec<DiversifierIndex>` but ended up going with storing only the max diversifier index in use since it's smaller and the point of the `DiversifierIndex` was to have a nice way to generate addresses iirc

Reply:
We probably want a way to track addresses, so maybe we should instead store a `Vec<String>` of labels, with `address_label[0]` having index `0`, etc?

Reply:
By storing the max `DiversifierIndex` in use, we can compute all the addresses in use via the `address_by_index` method on `ClientState` (deleted in the most recent commit), but it's more straightforward to store the encoded addresses, so I've made this change in a commit. The only downside is the wallet is no longer constant space but space is cheap